### DISCUSSION TITLE
Week 7: Instructional Notes

### DISCUSSION THREAD

**Discussion by bst3ck** [USU-ITLS-x265-23-summer/class-repo#143] [General]
As we have gone through this week's assignments, it was interesting to see the new notation formats introduced. I noticed that even though it was fewer characters to write, it was harder to read the functionality. Someone told me one time, the smarter you are, the fewer words you need to convey your ideas. I guess that is why it takes me so long to convey an idea. The same is true here. Even with fewer words, I prefer the older way of writing functions because I can follow the path easier.

As for the debugging lessons, this really made sense to me. I think I have made every one of those mistakes during our lessons this semester. I have had to use console.log tons in order to see why my code was not working. It seems like a secret weapon when trying to find out where I misplaced a variable :).

**Comment by metasean** [comment]
> Someone told me one time, the smarter you are, the fewer words you need to convey your ideas.

I've never heard a variation like this, but in general, the more of a domain expert that you are, the more likely it is that you'll have shortened variations of complex (relatively) concepts.  For example, when someone says, "sentence", you're likely to understand that means a combination of (at a minimum) a subject and a verb, but that also means that you understand what a subject is and what a verb is.  So, while "sentence" seems super simple and obvious to you at this point, remember all the things that you actually had to learn throughout K-12 to understand the larger concept.  I don't know about you, but I rarely explicitly think about the details of a sentence, e.g. to write this one, I didn't have to explicitly think about what my subject or verb were, it's simply become "intuitive".  In other words, we aren't "smarter", we've simply spent time building up domain knowledge.
